ast/ast.go

```diff
@@ -17,14 +17,22 @@ package ast
 
 import (
 	"io"
+	"strings"
 
 	"github.com/pingcap/parser/model"
 	"github.com/pingcap/parser/types"
 )
 
+// Recoverable can be restored to sql text
+type Recoverable interface {
+	// Restore returns the sql text from ast tree
+	Restore(sb *strings.Builder) error
```

Comment:
It seems that Restore does not need to return an error.

We can panic in the unimplemented Restore functions.

Reply:
For engineering consideration, we shouldn't painc to crash the process. and because of community cooperation, we add a error return to prevent unit test crash when some `Restore` is unimplemented

Reply:
We can use type assertion as I mentioned below.
From the engineering point of view, the definition of this useless return argument is unfriendly to the invoker since we must check the nonexistent error every time.